core/src/main/scala/algebra/ring/BoolRing.scala

```diff
@@ -0,0 +1,26 @@
+package algebra
+package ring
+
+import lattice.Bool
+
+/**
+ * A Boolean ring is a ring whose multiplication is idempotent, that is
+ * `a⋅a = a` for all elements ''a''. This property also implies `a+a = 0`
+ * for all ''a'', and `a⋅b = b⋅a` (commutativity of multiplication).
+ */
+trait BoolRing[A] extends Any with CommutativeRing[A] { self =>
+  override final def negate(x: A): A = x
+
+  def asBool: Bool[A] = new Bool[A] {
```

Comment:
might be nice to add the documentation to just mentio what the mapping is here (and == times, for instance).